engine/execution/result.go

```diff
@@ -0,0 +1,108 @@
+package execution
+
+import (
+	"github.com/onflow/flow-go/fvm/state"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+// CollectionExecutionResult holds aggregated artifacts (events, tx resutls, ...)
+// generated during collection execution
+type CollectionExecutionResult struct {
+	events                 flow.EventsList
```

Comment:
maybe just make these public fields.  then you don't need to define accessors.  same for CollectionAttestationResult

Reply:
Part of my current thinking for the upcoming PRs is to use the interfaces (ones that I temporarily keep in the `result` package) for components outside of the scope computation engine to subscribe to events instead of being dependent on execution.ComputationResult struct for testing and operations. then the current execution.ComputationResult could act as a collector of results inside the computation engine
```go
type ComputationManager interface {
	BlockExecutor
	BlockAttestor
	QueryResponder
} 
type BlockExecutor interface {
	ExecuteBlock (
	    ctx context.Context,
		block *entity.ExecutableBlock,
		snapshot state.StorageSnapshot,
	) (
		result.ExecutedBlock,
		error,
	)
}
type BlockAttestor interface { 
	AttestBlock(
	    ctx context.Context,
		eb result.ExecutedBlock,
	) (
	    result.AttestedBlock,
	    error, 
	)
} 
type QueryResponder interface { 
	GetAccount(
		ctx context.Context,
		addr flow.Address,
		header *flow.Header,
		snapshot state.StorageSnapshot,
	) (
		*flow.Account,
		error,
	)
	ExecuteScript(
		ctx context.Context,
		script []byte,
		arguments [][]byte,
		blockHeader *flow.Header,
		snapshot state.StorageSnapshot,
	) (
		[]byte,
		error,
	)
}

```

What do you think ?